src/adapter/src/coord/sequencer/inner.rs

```diff
@@ -1798,8 +1798,13 @@ impl Coordinator {
         session: &mut Session,
         plan: SetVariablePlan,
     ) -> Result<ExecuteResponse, AdapterError> {
-        let vars = session.vars_mut();
         let (name, local) = (plan.name, plan.local);
+
+        if &name == TRANSACTION_ISOLATION_VAR_NAME && session.transaction().contains_ops() {
```

Comment:
Should this and sequence_set_transaction both call some common function? I imagine other transaction modes in the future might also want to do this.

Reply:
Done!